utils/record/record.go

```diff
@@ -89,6 +94,39 @@ type FakeEventRecorder struct {
 	Events []string
 }
 
+func NewFakeApiFactory() api.Factory {
+	var (
+		settings = api.Settings{ConfigMapName: "my-config-map", SecretName: "my-secret", InitGetVars: func(cfg *api.Config, configMap *corev1.ConfigMap, secret *corev1.Secret) (api.GetVars, error) {
```

Comment:
where are the names `my-config-map` and `my-secret` coming from?

Same question on line 107, 113.

Reply:
Ideally they should be coming from controller.go
but as we are creating here FakeApiFactory for testing giving in the values here directly.. similar to that of notification engine.